dotCMS/src/main/java/com/dotmarketing/viewtools/DotTemplateTool.java

```diff
@@ -89,35 +95,59 @@ public static TemplateLayout themeLayout ( String themeInode ) throws DotDataExc
     public static TemplateLayout themeLayout ( String themeInode, Boolean isPreview ) throws DotDataException, DotSecurityException {
         String key = themeInode + isPreview;
         TemplateLayout layout = layoutCache.getIfPresent(key);
-        if(layout==null){
-        String title = null;
-        String drawedBody;
-        if ( UtilMethods.isSet( themeInode ) ) {
-            Identifier ident = APILocator.getIdentifierAPI().findFromInode( themeInode );
-            
-            Template template = APILocator.getTemplateAPI().findWorkingTemplate(ident.getId(), sysUser, false);
 
-            if ( !template.getInode().equals( themeInode ) ) {
-                template = (Template) InodeFactory.getInode( themeInode, Template.class );
+        if(layout==null) {
+            String title = null;
+            String drawedBody;
+            if (UtilMethods.isSet(themeInode)) {
+                Identifier ident = APILocator.getIdentifierAPI().findFromInode(themeInode);
+
+                Template template = APILocator.getTemplateAPI().findWorkingTemplate(ident.getId(), sysUser, false);
+
+                if (!template.getInode().equals(themeInode)) {
+                    template = (Template) InodeFactory.getInode(themeInode, Template.class);
+                }
+
+                drawedBody = ((Template) template).getDrawedBody();
+                title = template.getTitle();
+            } else {
+                drawedBody = (String) request.getAttribute("designedBody");
+                if (request.getAttribute("title") != null) {
+                    title = (String) request.getAttribute("title");
+                }
             }
 
-            drawedBody = ((Template) template).getDrawedBody();
-            title = template.getTitle();
-        } else {
-            drawedBody = (String) request.getAttribute( "designedBody" );
-            if ( request.getAttribute( "title" ) != null ) {
-                title = (String) request.getAttribute( "title" );
+            //Parse and return the layout for this template
+
+            try {
+                layout = getTemplateLayoutFromJSON(drawedBody);
+            } catch (IOException e) {
+                layout = DesignTemplateUtil.getDesignParameters(drawedBody, isPreview);
             }
-        }
 
-        //Parse and return the layout for this template
-        layout = DesignTemplateUtil.getDesignParameters( drawedBody, isPreview );
-        layout.setTitle( title );
-        layoutCache.put(key, layout);
+            layout.setTitle(title);
+            layoutCache.put(key, layout);
         }
+
         return layout;
     }
 
+    private static TemplateLayout getTemplateLayoutFromJSON(String json)  throws IOException{
+        return JsonTransformer.mapper.readValue(json, TemplateLayout.class);
+
+    }
+
+    private static String getTemplateBody(String themeInode) throws DotDataException, DotSecurityException {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "getTemplateBody" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUnusedPrivateMethod)